peer/brontide_test.go

```diff
@@ -1589,3 +1591,52 @@ func TestRemovePendingChannel(t *testing.T) {
 
 	require.NoError(t, err)
 }
+
+// TestHandlePeerStorage tests handling peer storage message.
+func TestHandlePeerStorage(t *testing.T) {
+	t.Parallel()
+	rt := require.New(t)
+
+	peerDataStore := newMockDataStore()
+	privkey, err := btcec.NewPrivateKey()
+	rt.NoError(err, "error while generating private key.")
+
+	// Create a test brontide.
+	dummyConfig := Config{
+		DataStoreForPeer: peerDataStore,
+		Addr: &lnwire.NetAddress{
+			IdentityKey: privkey.PubKey(),
+		},
+	}
+	peer := NewBrontide(dummyConfig)
+	peer.outgoingQueue = make(chan outgoingMsg)
+
+	// Call the method.
+	testBackup := []byte{0x9c, 0x40, 0x1, 0x2, 0x3}
+	go peer.handlePeerStorageMessage(&lnwire.PeerStorage{
+		Blob: testBackup,
+	})
```

Comment:
Depends on the decision we take here, 
https://github.com/lightningnetwork/lnd/pull/8490/commits/747b152d995532850935acb6b7ebbc5862a9f899#diff-8986cde7a8ea16ac62b7e29ee740f3f307907e60c74818cb05125a3508c60bb9R4125

Should it block till it completes or not?

Reply:
> Depends on the decision we take here

Use `p.queueMessage` instead there

Reply:
How is it better than this?

Reply:
using SendMessage from inside the Brontide can create synchronization issues. You are trying to deal with that using this goroutine forking in this test. In general we should not be using writeMessage directly inside the Brontide. It can lead to deadlocks.

Reply:
I don't think it is because I am calling it inside the package. I think it should have the same behaviour calling it in and out of the package. The reason I have to run on a different thread is because of this line: https://github.com/Chinwendu20/lnd/blob/cd566eb0973240667d37823aaee2b5a61a012173/peer/brontide.go#L3507-L3518
That is not necessarily a bad thing as it is required in some scenarios and this might be one of them, because I set sync to true.  I could skip it by sending skip to false.

Reply:
You are correct, in this case you can leave the API call but I'd set `sync` to `false`

Reply:
Could you share why you prefer this?

Reply:
you can just buffer the output channel by 1 and then there is no need for the goroutine

Reply:
Thanks, how is this better than the goroutine?

Reply:
you guys where asking about how to make this a synchronous test. Synchronous tests are generally easier to read

Reply:
> Thanks, how is this better than the goroutine?

We should avoid forking new goroutines whenever possible. It should be treated as a last-resort. More threads -> more problems.